File: 2342_max_sum_of_a_pair_with_equal_sum_of_digits.cpp
class Solution {
public:
    int zyx(int n) {
        int a = 0;
        while(n) {
            a += n % 10;
            n /= 10;
        }
        return a;
    }
    
    int maximumSum(vector<int>& v) {
        int res = -1, z[82] = {};
        for(int b : v) {
            int c = zyx(b);
            if(z[c]) res = max(res, z[c] + b);
            z[c] = max(z[c], b);
        }
        return res;
    }
};

File: 2342_max_sum_of_a_pair_with_equal_sum_of_digits.java
class Solution {
    int zap(int n) {
        int s = 0;
        while (n > 0) {
            s += n % 10;
            n /= 10;
        }
        return s;
    }

    public int maximumSum(int[] arr) {
        int res = -1, z[] = new int[82];
        for (int x : arr) {
            int d = zap(x);
            if (z[d] > 0) res = Math.max(res, z[d] + x);
            z[d] = Math.max(z[d], x);
        }
        return res;
    }
}
